src/main/java/EPparser/CommandParser.java

```diff
@@ -0,0 +1,237 @@
+                if (command.isValidCommand()) {
+                    CommandStack.pushCmd(mc);
+                }
+                break;
+            case yes:
+                System.out.println("Yes");
+                YesCommand yc = new YesCommand(UIController);
+                yc.initCommand(CommandArr, Command, command.getSubRootCommand());
+                if (command.isValidCommand()) {
+                    CommandStack.pushCmd(yc);
+                }
+                break;
+            case add:
+                System.out.println("Yes");
+                AddCommand wc = new AddCommand(UIController);
+                wc.initCommand(CommandArr, Command, command.getSubRootCommand());
+                if (command.isValidCommand()) {
+                    CommandStack.pushCmd(wc);
+                }
+                break;
+            case set:
+                System.out.println("Set");
+                SetCommand stc = new SetCommand(UIController);
+                stc.initCommand(CommandArr, Command , command.getSubRootCommand());
+                if (command.isValidCommand()) {
+                    CommandStack.pushCmd(stc);
+                }
+                break;
+            case playlist:
+                System.out.println("Playlist");
+                PlaylistCommand pc = new PlaylistCommand(UIController);
+                pc.initCommand(CommandArr, Command , command.getSubRootCommand());
+                if (command.isValidCommand()) {
+                    CommandStack.pushCmd(pc);
+                }
+                break;
+            case preference:
+                System.out.println("Preference");
+                PreferenceCommand pfc = new PreferenceCommand(UIController);
+                pfc.initCommand(CommandArr, Command , command.getSubRootCommand());
+                if (command.isValidCommand()) {
+                    CommandStack.pushCmd(pfc);
+                }
+                break;
+            case restriction:
+                System.out.println("Restriction");
+                RestrictionCommand rc = new RestrictionCommand(UIController);
+                rc.initCommand(CommandArr, Command, command.getSubRootCommand());
+                if (command.isValidCommand()) {
+                    CommandStack.pushCmd(rc);
+                }
+            case remove:
+                System.out.println("REmove");
+                RemoveCommand removec = new RemoveCommand(UIController);
+                removec.initCommand(CommandArr , Command);
+
+                if (command.isValidCommand()) {
+                    CommandStack.pushCmd(removec);
+                }
+                break;
+            default:
+                CommandPair pair = CommandDebugger.commandSpellChecker(CommandArr , COMMANDKEYS.none , UIController);
+                ((MovieHandler) UIController).setFeedbackText("Sorry we are unable to process your command. Please check help for more details!");
+        }
+    }
```

Comment:
This method is way too long. Find a way to shorten it by abstracting out lower level codes.